node-tests/unit/commands/exam-test.js

```diff
@@ -97,29 +105,116 @@ describe('ExamCommand', function() {
+      const randomStub = sinon.stub(Math, 'random').returns('  random_seed');
       return command.run({ random: '' }).then(function() {
         assert.equal(called.testRunOptions.query, 'seed=random_seed');
         randomStub.restore();
       });
     });
+  });
+
+  describe('_appendParamToBaseUrl', function() {
+    function appendParamToBaseUrl(commandOptions, baseUrl) {
+      const project = new MockProject();
+      project.isEmberCLIProject = function() { return true; };
 
-    it('should set \'weighted\' on the query option', function() {
-      return command.run({ split: 2, partition: [2], weighted: true }).then(function() {
-        assert.equal(called.testRunOptions.query, '_split=2&_partition=2&_weighted');
+      const command = new ExamCommand({
+        project: project,
+        tasks: {}
       });
+
+      command.validator = new TestOptionsValidator(commandOptions);
+
+      return command._appendParamToBaseUrl(commandOptions, baseUrl);
+    }
+
+    it('should add `split` when `split` option is used.', function() {
+      const appendedUrl = appendParamToBaseUrl({ split: 3 }, 'tests/index.html?hidepassed');
+
+      assert.deepEqual(appendedUrl, 'tests/index.html?hidepassed&split=3');
     });
-  });
+
+    it('should add `split` when `split` and `parallel` option are used.', function() {
+      const appendedUrl = appendParamToBaseUrl({ split: 5, parallel: true }, 'tests/index.html?hidepassed');
+
+      assert.deepEqual(appendedUrl, 'tests/index.html?hidepassed&split=5');
+    });
+
+    it('should add `loadBalance` when `load-balance` option is used.', function() {
+      const appendedUrl = appendParamToBaseUrl({ loadBalance: 2 }, 'tests/index.html?hidepassed');
+
+      assert.deepEqual(appendedUrl, 'tests/index.html?hidepassed&loadBalance');
+    });
+
+    it('should add `split`, `loadBalance`, and `partition` when `split`, `loadBalance`, and `partition` are used.', function() {
+      const appendedUrl = appendParamToBaseUrl({ split: 5, partition: [1,2,3], loadBalance: 2 }, 'tests/index.html?hidepassed');
+
+      assert.deepEqual(appendedUrl, 'tests/index.html?hidepassed&split=5&loadBalance&partition=1&partition=2&partition=3');
+    });
+
+    it('should add `loadBalance` when `replay-execution` and `replay-browser` are used.', function() {
+      const appendedUrl = appendParamToBaseUrl({ replayExecution: 'test-execution-0000000.json', replayBrowser: [1, 2] }, 'tests/index.html?hidepassed');
+
+      assert.deepEqual(appendedUrl, 'tests/index.html?hidepassed&loadBalance');
+    });
+  })
+
+  describe('_getCustomBaseUrl', function() {
+    function getCustomBaseUrl(config, options) {
+      const project = new MockProject();
+      project.isEmberCLIProject = function() { return true; };
+
+      const command = new ExamCommand({
+        project: project,
+        tasks: {}
+      });
+
+      command.validator = new TestOptionsValidator(options);
+
+      return command._getCustomBaseUrl(config, options)
```

Comment:
Testing private API in tests suggests some abstraction violation. Either it should be standalone and tested, or maybe it should be simply considered public API.